src/deprecates.jl

```diff
@@ -26,3 +26,10 @@ end
 @deprecate scattermat_zm(x::DenseMatrix, wv::AbstractWeights, dims::Int) scattermat_zm(x::DenseMatrix, wv::AbstractWeights, dims=dims)
 @deprecate mean!(R::AbstractArray, A::AbstractArray, w::AbstractWeights, dims::Int) mean!(R, A, w, dims=dims)
 @deprecate mean(A::AbstractArray{T}, w::AbstractWeights{W}, dims::Int) where {T<:Number,W<:Real} mean(A, w, dims=dims)
+
+@deprecate wquantile(v::RealVector, w::AbstractWeights{<:Real}, p::RealVector) quantile(v, w, p)
+@deprecate wquantile(v::RealVector, w::AbstractWeights{<:Real}, p::Number) quantile(v, w, [p])[1]
+@deprecate wquantile(v::RealVector, w::RealVector, p::RealVector) quantile(v, weights(w), p)
+@deprecate wquantile(v::RealVector, w::RealVector, p::Number) quantile(v, weights(w), [p])[1]
+@deprecate wmedian(v::RealVector, w::AbstractWeights{<:Real}) median(v, w)
+@deprecate wmedian(v::RealVector, w::RealVector) median(v, weights(w))
```

Comment:
This makes me realize that we probably shouldn't accept `Weights` objects in `quantile` and `median` since their meaning is ambiguous: they could be frequency weights, or another type of weights. So here we'd better recommend using `pweights`or `fweights` (whichever is closer to the current behavior of `wmedian`).

Reply:
Isn't it ok to assume that default weights are probability weights, as it is now? Honestly, most of the time these differences don't matter, and I don't want users starting to feel overwhelmed by the exact weight type they should use.

In Stata:
> Each command has its own idea of the "natural" kind of weight.  The command will tell you what kind of weight it is assuming and perform the request as if you specified that kind of weight.

Reply:
I'd like that too. I don't think it's too late.

Reply:
I don't really like commands that talk to the user. Either a command succeeds and it should just do what is requested, or it doesn't and it should indicate a way to make it succeed. Printing warnings during normal operation is just annoying, we'd better explain how to do things properly. Adding `f`, `p` or `a` in front of `weights` isn't that costly, and anyway if users doesn't know what letter to use they won't understand the warning (and likely do something incorrect).

Also I don't think is common for Julia functions to print messages like that.

Reply:
I was talking about the second point ;)